### PR TITLE
docs: add Pinecone integration section

### PR DESCRIPTION
**Pull request by jirispilka**
This is a draft of the Pinecone integration documentation. Pinecone requires this to progress further with the partnership. As discussed with @jancurn, we should have it anyway.

@TC-MO, can you please check the structure, files, and grammar (and if I added everything required for a new page)?
Can we publish it on Tuesday or Wednesday? 
Basically, we need a link that we can share. We can improve the documentation after that.

### CODE REVIEW

File: sources/platform/integrations/pinecone.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+Pinecone is a managed vector database that allows users to store and query dense vectors for AI applications such as recommendation systems, semantic search, and retrieval augmented generation.
+
+## How to integrate Apify with Pinecone?
+
+This example focuses on integrating Apify with Pinecone using the Apify Actor.
+
+Before we start, ensure you have a [Pinecone database](https://www.pinecone.io/) set up.
+Create a Pinecone index and obtain the Pinecone API token.
+Additionally, you need an [OpenAI API key](https://openai.com/index/openai-api/) and [Apify API token](https://docs.apify.com/platform/integrations/api#api-token).
```
Comment (jancurn) [inline sources/platform/integrations/pinecone.md:24]: Pls let's mention why the "OpenAI API key" is needed

File: sources/platform/integrations/pinecone.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+Next, install the apify-client to interact with the Apify API:
```
Comment (jancurn) [inline sources/platform/integrations/pinecone.md:26]: Is there a way to set this up manually in UI, using the actor-to-actor integration? If yes, please can we explain it here? e.g. have few sections how to set it up, one in Python, second in UI, another (in the future) with JS.

File: sources/platform/integrations/pinecone.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+**Learn how to integrate Apify with Pinecone to feed data crawled from the web into the Pinecone vector database.**
+
+---
+
+> For more information on Pinecone, visit its [documentation](https://docs.pinecone.io).
+
+## What is Pinecone?
+
+Pinecone is a managed vector database that allows users to store and query dense vectors for AI applications such as recommendation systems, semantic search, and retrieval augmented generation.
```
Comment (jancurn) [inline sources/platform/integrations/pinecone.md:16]: Let's make the first mention of Pinecone a link to https://www.pinecone.io/

File: sources/platform/integrations/pinecone.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+## How to integrate Apify with Pinecone?
+
+This example focuses on integrating Apify with Pinecone using the Apify Actor.
```
Comment (jancurn) [inline sources/platform/integrations/pinecone.md:20]: ## How to integrate Apify with Pinecone?

The Apify integration for Pinecone lets you export results from Actors and Dataset items into a specific Pinecone vector index. Note that you also need an [OpenAI platform](https://platform.openai.com/docs/overview) account with an API key in order to compute the text embeddings.

The integration is facilitated using the [Pinecone integration](https://apify.com/jan.turon/pinecone-integration) Actor.

## Examples

These examples use the [Website Content Crawler](https://apify.com/apify/website-content-crawler), an Actor that can deeply crawl websites, clean their HTML by removing a cookies modal, footer, or navigation, and then transform the HTML to Markdown, which can be used as training data for AI models or for feeding LLM and generative AI applications with web content.

### User interface

...

### Python

....

File: sources/platform/integrations/pinecone.md (hunk 2 above)
Comment (jirispilka) [reply sources/platform/integrations/pinecone.md:26]: Thanks. Good point. I added the UI section with screenshots.

File: sources/platform/integrations/pinecone.md (hunk 4 above)
Comment (jirispilka) [reply sources/platform/integrations/pinecone.md:20]: Thanks, added.

**Comment by TC-MO** [comment]
I've made my changes to the document:

* reworked the whole structure for better logical flow
* rewritten prose for readability, clarity & conciseness
* renamed headings and removed punctuation from them to adhere to microsoft style guide
* added addition ToC levels through front-matter so that each integration method appears in ToC
* added admonitions for better visual distinction

Let me know if these changes make sense to you.

Regarding the lychee, we can disregard all but 2 of them which as all 404 refer to this document ( This is due to the way lychee has to check our content against live website), if nothing can be helped with the other two we can disregard this check I'll update in an hour

**Comment by TC-MO** [comment]
Lychee update:

As far as I'm concerned we can disregard it failing.

* 404's are expected
* other errors seems to not be our fault , and I'll try to address them in separate PR but it will also involve combing through content and trying to find similar link or reworking the content so it works without links, so it shouldn't stop this PR

**Comment by jirispilka** [comment]
> I've made my changes to the document:
> 
> * reworked the whole structure for better logical flow
> * rewritten prose for readability, clarity & conciseness
> * renamed headings and removed punctuation from them to adhere to microsoft style guide
> * added addition ToC levels through front-matter so that each integration method appears in ToC
> * added admonitions for better visual distinction
> 
> Let me know if these changes make sense to you.
> 
> Regarding the lychee, we can disregard all but 2 of them which as all 404 refer to this document ( This is due to the way lychee has to check our content against live website), if nothing can be helped with the other two we can disregard this check I'll update in an hour

Thanks a lot!

**Review by TC-MO — APPROVED** [review]
If my changes are ok, then approve from me

**Review by jancurn — APPROVED** [review]
Great job Jirka

**Pull request merged**